spec/close_milestone_action_spec.rb

```diff
@@ -1,105 +1,86 @@
-    end
+  it 'raises an error when the milestone is not found or does not exist' do
+    allow(client).to receive(:list_milestones).and_return([])
+    expect { run_described_fastlane_action(default_params) }.to raise_error(FastlaneCore::Interface::FastlaneError, 'Milestone 10.1 not found.')
+  end
 
-    it 'properly passes the repository all the way down to the Octokit::Client to list the milestones' do
-      allow(Octokit::Client).to receive(:new).and_return(client)
-      expect(client).to receive(:list_milestones).with(test_repository)
-      run_described_fastlane_action(mock_params)
+  describe 'Calling the Action validates input' do
+    it 'raises an error if no `GITHUB_TOKEN` environment variable nor parameter `:github_token` is present' do
+      ENV['GITHUB_TOKEN'] = nil
+      expect { run_action_without_key(:github_token) }.to raise_error(FastlaneCore::Interface::FastlaneError, "No value found for 'github_token'")
     end
-  end
 
-  describe 'update_milestone' do
-    let(:test_repository) { 'test-repository' }
-    let(:test_milestone_number) { '1234' }
-    let(:mock_params) do
-      {
-        repository: test_repository,
-        milestone: '10',
-        github_token: 'Test-GithubToken-1234'
-      }
+    it 'raises an error if no `GHHELPER_REPOSITORY` environment variable nor parameter `:repository` is present' do
+      expect { run_action_without_key(:repository) }.to raise_error(FastlaneCore::Interface::FastlaneError, "No value found for 'repository'")
     end
-    let(:client) do
-      instance_double(
-        Octokit::Client,
-        list_milestones: [{ title: '10.1', number: test_milestone_number }],
-        update_milestone: nil,
-        user: instance_double('User', name: 'test'),
-        'auto_paginate=': nil
-      )
+
+    it 'raises an error if no `GHHELPER_MILESTONE` environment variable nor parameter `:milestone` is present' do
+      expect { run_action_without_key(:milestone) }.to raise_error(FastlaneCore::Interface::FastlaneError, "No value found for 'milestone'")
     end
 
-    it 'properly passes the parameters all the way down to Octokit::Client' do
-      allow(Octokit::Client).to receive(:new).and_return(client)
-      expect(client).to receive(:update_milestone).with(test_repository, test_milestone_number, { state: 'closed' })
-      run_described_fastlane_action(mock_params)
+    it 'raises an error if `milestone:` parameter is passed as Integer' do
+      expect { run_action_with(:milestone, 10) }.to raise_error "'milestone' value must be a String! Found Integer instead."
     end
   end
+
+  def run_action_without_key(key)
+    run_described_fastlane_action(default_params.except(key))
+  end
+
+  def run_action_with(key, value)
+    values = default_params
+    values[key] = value
+    run_described_fastlane_action(values)
+  end
+
+  def default_params
+    {
+      repository: test_repository,
+      milestone: test_milestone[:title],
+      github_token: test_token
+    }
+  end
```

Comment:
That confused me a bit tbh 🤔. 

Because, If I put those in a `let(:default_params)` and then remove params from it, like I'm doing on the `run_action_without(key)`, as `let` means that this is a constant by convention and as you [stated here](https://github.com/wordpress-mobile/release-toolkit/pull/425#discussion_r1013252381), that [although is allowed is not right](https://github.com/wordpress-mobile/release-toolkit/pull/425#discussion_r1013253747) and should be avoided. 

I thought if I put the parameters again as a `let` and then do those operations, I will be breaking the conventions again. 😅

Reply:
`except` is not a mutation method tho (unlike `[]`)